pybossa/cache/helpers.py

```diff
@@ -192,11 +194,11 @@ def n_available_tasks_for_user(project, user_id=None, user_ip=None):
                AND id NOT IN
                (SELECT task_id FROM task_run WHERE
                project_id=:project_id AND user_id=:user_id)
-               AND (user_pref IS NULL OR {}) {} ;
```

Comment:
Not sure why  `user_pref IS NULL ` is a condition here? We are doing user_pref_scheduler and I suppose  non-preference tasks should be assigned to non-preference users.